romancal/resample/resample_utils.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+import numpy as np
+
+from romancal.assign_wcs.utils import wcs_from_footprints, wcs_bbox_from_shape
+
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def make_output_wcs(input_models, ref_wcs=None,
+                    pscale_ratio=None, pscale=None, rotation=None, shape=None,
+                    crpix=None, crval=None):
+    """ Generate output WCS here based on footprints of all input WCS objects
+    Parameters
+    ----------
+    input_models : list of `~jwst.datamodel.DataModel`
+        Each datamodel must have a ~gwcs.WCS object.
+
+    pscale_ratio : float, optional
+        Ratio of input to output pixel scale. Ignored when ``pscale`` is provided.
+
+    pscale : float, None, optional
+        Absolute pixel scale in degrees. When provided, overrides
+        ``pscale_ratio``.
+
+    rotation : float, None, optional
+        Position angle of output image’s Y-axis relative to North.
+        A value of 0.0 would orient the final output image to be North up.
+        The default of `None` specifies that the images will not be rotated,
+        but will instead be resampled in the default orientation for the camera
+        with the x and y axes of the resampled image corresponding
+        approximately to the detector axes.
+
+    shape : tuple of int, None, optional
+        Shape of the image (data array) using ``numpy.ndarray`` convention
+        (``ny`` first and ``nx`` second). This value will be assigned to
+        ``pixel_shape`` and ``array_shape`` properties of the returned
+        WCS object.
+
+    crpix : tuple of float, None, optional
```

Comment:
A minor nit, Is this really crval and crpix? The roman wcs has ra_ref, dec_ref and v2 & v3_ref. Maybe we should change crpix and crval to something more generic? Maybe ref_pixel and ref_coord? Anyone else have an opinion on this?

Reply:
Done.